Ci test docs

ahmadtourei opened the pull request:
<!--
Thanks for contributing to DASCore, community contributions are most welcomed!

Before contributing, please read through the [contributors doc](https://dascore.org/contributing/contributing.html)

Before making big changes to the code or adding large complex features, it is a good idea to
[open a discussion](https://github.com/DASDAE/dascore/discussions). Don't hesitate to ask a question or for
help if something isn't clear.
-->

## Description
This PR adds GitHub workflows to build docs in PRs with the "documentation" label.
<!--
Please describe your PR here. What problem are you trying to solve, or what feature are you adding?

Also link any relevant issues/discussions (this can be done using the issue/discussion number preceded by a
pound sign, e.g. `#12` without the backticks)
-->

## Checklist

I have (if applicable):

- [ ] referenced the GitHub issue this PR closes.
- [ ] documented the new feature with docstrings or appropriate doc page.
- [ ] included a test. See [testing guidelines](https://dascore.org/contributing/testing.html).
- [ ] your name has been added to the contributors page (docs/contributors.md).
- [ ] added the "ready_for_review" tag once the PR is ready to be reviewed.

ahmadtourei commented:
Although I added "documentation" label to this PR (and then also re-ran the workflow tests), the `TestCode/build-docs` is still skipped. Maybe this will be fixed in the next PRs?

d-chambers commented:
> Although I added "documentation" label to this PR (and then also re-ran the workflow tests), the TestCode/build-docs is still skipped. Maybe this will be fixed in the next PRs?

It appears something may be broken with the action triggering the PR based on the tag. I think the comments on [this SO post](https://stackoverflow.com/a/59588725/3645626) might be helpful. 

Actually, thinking more about it, I am inclined to put this in it's own action. `TestCode` runs on both PRs and when code is merged into master. In the latter case, the docs would be built twice, which is a waste of CI resources. 

So, I suggest making a new workflow called `test_doc_build.yml`. Maybe something like this? 

```
name: TestDocBuild

on:
  pull_request:
    types: [labeled, synchronize]

jobs:
  conditional-job:
    if: |
      (github.event.action == 'labeled' && github.event.label.name == 'documentation')
      || (github.event.action == 'synchronize' && contains(github.event.pull_request.labels.*.name, 'documentation'))
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-tags: 'true'

      - uses: ./.github/actions/mamba-install-dascore
        with:
          python-version: "3.12"
          environment-file: './.github/doc_environment.yml'

      - uses: ./.github/actions/build-docs

      - uses: actions/upload-artifact@v4
        with:
          name: documentation_build_test
          path: ./docs/_site
          retention-days: 1 
 ```

If this is right, it will run under two conditions: 1) when a new commit is pushed to the PR AND the PR has the `documentation` label. 2) When the `documentation` label is added to the PR.

ahmadtourei commented on .github/workflows/labeler.yml:
```diff
@@ -0,0 +1,19 @@
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@v4 # pulls the PR code
```
Yeah I don't think it's needed.

ahmadtourei commented on .github/workflows/runtests.yml:
```diff
@@ -66,8 +66,32 @@ jobs:
+
+      - uses: ./.github/actions/build-docs
 
+      - name: publish docs to netlify
+        shell: bash -l {0}
+        env:
+          QUARTO_PRINT_STACK: true
+          NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
+        run: |
+            quarto publish docs --no-prompt --no-browser
+            
```
Agreed. Just removed it.

ahmadtourei commented:
> > Although I added "documentation" label to this PR (and then also re-ran the workflow tests), the TestCode/build-docs is still skipped. Maybe this will be fixed in the next PRs?
> 
> It appears something may be broken with the action triggering the PR based on the tag. I think the comments on [this SO post](https://stackoverflow.com/a/59588725/3645626) might be helpful.
> 
> Actually, thinking more about it, I am inclined to put this in it's own action. `TestCode` runs on both PRs and when code is merged into master. In the latter case, the docs would be built twice, which is a waste of CI resources.
> 
> So, I suggest making a new workflow called `test_doc_build.yml`. Maybe something like this?
> 
> ```
> name: TestDocBuild
> 
> on:
>   pull_request:
>     types: [labeled, synchronize]
> 
> jobs:
>   test_build_docs:
>     if: |
>       (github.event.action == 'labeled' && github.event.label.name == 'documentation')
>       || (github.event.action == 'synchronize' && contains(github.event.pull_request.labels.*.name, 'documentation'))
>     runs-on: ubuntu-latest
>     steps:
>       - uses: actions/checkout@v4
>         with:
>           fetch-tags: 'true'
> 
>       - uses: ./.github/actions/mamba-install-dascore
>         with:
>           python-version: "3.12"
>           environment-file: './.github/doc_environment.yml'
> 
>       - uses: ./.github/actions/build-docs
> 
>       - uses: actions/upload-artifact@v4
>         with:
>           name: documentation_build_test
>           path: ./docs/_site
>           retention-days: 1 
> ```
> 
> If this is right, it will run under two conditions: 1) when a new commit is pushed to the PR AND the PR has the `documentation` label. 2) When the `documentation` label is added to the PR.

Yes, it makes more sense to have this under a new workflow test. Just added it.  However, shouldn't we have `contains(github.event.pull_request.labels.*.name, 'documentation')` in the first if statement?
```python
      (github.event.action == 'labeled' && contains(github.event.pull_request.labels.*.name, 'documentation'))
      || (github.event.action == 'synchronize' && contains(github.event.pull_request.labels.*.name, 'documentation'))
```

d-chambers commented:
Actually no, I just added the CI tag and it triggered the doc build.

We want it to only trigger it if the `documentation` tag is added OR the `documentation` tag already exists and a new commit is pushed to the PR.

ahmadtourei commented:
> github.event.label.name == 'documentation'

Gotcha, thanks for clarifying. Fun little CI project :)
